common/djangoapps/track/backends/tests/test_oli.py

```diff
@@ -0,0 +1,273 @@
+"""
+Assert that the OLI Analytics Backend transmits events properly
+"""
+from __future__ import absolute_import
+
+import json
+from mock import patch, Mock
+import logging
+
+from django.conf import settings
+from django.test import TestCase
+from south.utils import datetime_utils as datetime
+
+from student.tests.factories import UserFactory
+from track.backends.oli import OLIAnalyticsBackend
+
+
+LOG = logging.getLogger(__name__)
+
+
+class TestOLIBackend(TestCase):
+    """
+    Assert that the OLI Analytics Backend transmits events properly
+
+    OLI should only process 'problem_check' event types.
+    Course IDs must be part of the whitelisted set.
+    Most event data like user_agent is arbitrary and can be changed.
+    """
+
+    def setUp(self):
+        """
+        Initialize OLI Backend tests
+        """
+        self.patcher = patch('track.backends.oli.OAuth1Session.put', mock_put)
+        self.addCleanup(self.patcher.stop)
+        self.patcher.start()
+        course_ids = {
+            'edX/DemoX/Demo_Course',
+            'edX/DemoX/Demo_Course1',
+        }
+        self.backend = OLIAnalyticsBackend(
+            url='https://example.com/',
+            path='path',
+            key='key',
+            secret='secret',
+            course_ids=course_ids,
+        )
+        self.user = UserFactory.create()
+        self.user.save()
+
+    def test_oli_backend_correct_event_type(self):
+        """
+        Assert that problem_check events are sent
+        """
+        # pylint: disable=line-too-long
+        event = {
+            'username': 'verified',
+            'event_type': 'problem_check',
+            'ip': '10.0.2.2',
+            'agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.124 Safari/537.36',
+            'host': 'precise64',
+            'referer': 'https://{site_name}/courses/edX/DemoX/Demo_Course/courseware/interactive_demonstrations/basic_questions/'.format(
```

Comment:
Nit: Since we're mocking every HTTP request anyway, we probably don't need to parameterize this; we could just hard-code it to `example.com`.